File: Linkpoint/src/main/java/com/linkpoint/slproto/messages/UnsubscribeLoad.java
package com.linkpoint.slproto.messages;

import com.linkpoint.slproto.SLMessage;
import java.nio.ByteBuffer;

public class UnsubscribeLoad extends SLMessage {
    public UnsubscribeLoad() {
        this.zeroCoded = false;
    }

    public int CalcPayloadSize() {
        return 4;
    }

    public void Handle(SLMessageHandler sLMessageHandler) {
        sLMessageHandler.HandleUnsubscribeLoad(this);
    }

    public void PackPayload(ByteBuffer byteBuffer) {
        byteBuffer.putShort(-1);
        byteBuffer.put((byte) 0);
        byteBuffer.put((byte) 8);
    }

    public void UnpackPayload(ByteBuffer byteBuffer) {
    }
}

File: Linkpoint/src/main/java/com/linkpoint/slproto/messages/UnsubscribeLoad.kt
package com.linkpoint.slproto.messages

import com.linkpoint.slproto.SLMessage
import java.nio.ByteBuffer

class UnsubscribeLoad : SLMessage() {
    init {
        zeroCoded = false
    }

    override fun CalcPayloadSize(): Int = 4

    override fun Handle(handler: SLMessageHandler) {
        handler.HandleUnsubscribeLoad(this)
    }

    override fun PackPayload(buffer: ByteBuffer) {
        buffer.putShort(-1)
        buffer.put(0.toByte())
        buffer.put(8.toByte())
    }

    override fun UnpackPayload(buffer: ByteBuffer) {}
}